bionic/deriver.py

```diff
@@ -188,12 +188,12 @@ def _initialize_task_state(self, task_state):
             dep_state.children.append(task_state)
 
         # All names in this task should point to the same provider.
-        (task_state.provider,) = set(
+        (task_state.provider) = set(
```

Comment:
ah looks like black added brackets on the left side as well which weren't present before. Removing the comma before brackets made a lot of these variables a set which does not work anymore. I guess you will have to revert the `,)` to `)` change 😞.